src/pytesmo/validation_framework/adapters.py

```diff
@@ -314,3 +314,51 @@ def read_ts(self, *args, **kwargs):
     def read(self, *args, **kwargs):
         data = super(AnomalyClimAdapter, self).read(*args, **kwargs)
         return self.calc_anom(data)
+
+
+class ColumnCombAdapter(BasicAdapter):
+    """
+    Takes the pandas DataFrame that the read_ts or read method of the instance
+    returns and applies a function to merge multiple columns into one.
+    E.g. when there are 2 Soil Moisture parameters in a dataset that should be
+    averaged on reading. Will add one additional column to the input data frame.
+    """
+
+    def __init__(self, cls, func, func_kwargs=None, columns=None, new_name='merged'):
+        """
+        Parameters
+        ----------
+        cls : class instance
+            Must have a read_ts or read method returning a pandas.DataFrame
+        func: Callable
+            Will be applied to dataframe columns using pd.DataFrame.apply(..., axis=1)
+            additional kwargs for this must be given in func_kwargs,
+            e.g. pd.DataFrame.mean
+        func_kwargs : dict, optional (default: None)
+            kwargs that are passed to method
+        columns: list, optional
+            columns in the dataset that are combined.
```

Comment:
Can you add in the description that in case `columns=None`, all columns are used?